tiled/catalog/adapter.py

```diff
@@ -921,26 +926,27 @@ async def delete_tree(self, external_only=True):
                         "If you want to delete them, pass external_only=False."
                     )
             else:
-                sel_int_asset_statement = select(
-                    orm.Asset.data_uri, orm.Asset.is_directory
-                ).filter(
-                    orm.Asset.data_sources.any(
-                        orm.DataSource.management != Management.external
-                    )
-                )
-                for condition in conditions:
-                    sel_int_asset_statement.filter(condition)
-                int_assets = (await db.execute(sel_int_asset_statement)).all()
-                for data_uri, is_directory in int_assets:
-                    delete_asset(data_uri, is_directory)
+                raise NotImplementedError
             # TODO Deal with Assets belonging to multiple DataSources.
```

Comment:
I spent some time working through this, and it's quite tricky. Since this has no use cases for us yet, I propose to defer this (and open an issue) without letting it holdup the fix for the common case.